virtcontainers/factory/template/template.go

```diff
@@ -100,35 +104,44 @@ func (t *template) createTemplateVM(ctx context.Context) error {
 	config.HypervisorConfig.BootFromTemplate = false
 	config.HypervisorConfig.MemoryPath = t.statePath + "/memory"
 	config.HypervisorConfig.DevicesStatePath = t.statePath + "/state"
+	// template vm uses builtin proxy
+	config.ProxyType = templateProxyType
 
 	vm, err := vc.NewVM(ctx, config)
 	if err != nil {
 		return err
 	}
 	defer vm.Stop()
 
-	err = vm.Pause()
-	if err != nil {
+	if err = vm.Disconnect(); err != nil {
 		return err
 	}
 
-	err = vm.Save()
-	if err != nil {
+	// Sleep a bit to let the agent grpc server clean up
+	time.Sleep(templateWaitForAgent)
```

Comment:
Why a sleep here? Any way we could provide some more robust way of actively waiting?

Reply:
As I replied to @jodh-intel here (https://github.com/kata-containers/runtime/pull/684#issuecomment-419480476), it is an optimization worth investing.

OTOH, I don't see any possible active waiting method since we cannot connect to the agent to make sure it is up.

Reply:
Ok now I understand what you're doing here. I understand this is out of the regular path since it's about VM template creation that might happen hours before the creation of an actual container.
And this second makes sure the agent is already in a proper waiting state for when the template is going to be used.
That's hard to think how of a better way, I understand...
Could you open an issue to see if we can think about something else after this got merged.

Reply:
Sure, I'll open an issue for this but I don't think there is an easy way to do it and TBF I don't think it worth much effort to optimize the sleep time since it only happens once.